devito/passes/clusters/blocking.py

```diff
@@ -313,11 +316,15 @@ def decompose(ispace, d, block_dims):
             if bd._depth < i._depth:
                 relations.append((bd, i))
 
+    # Drop redundant relations
+    relations = set(relations) - {k for n, k in it.product(relations, repeat=2)
```

Comment:
Simplifies the connection graph.
Reduces from:
`
[(y0_blk0, y), (x0_blk0, y0_blk0, z), (x0_blk0, y, z), (t, x0_blk0, y0_blk0, z), (t, x0_blk0, y, z), (t, x, y, z), (x, y, z), (time, t), (x0_blk0, x), (time, x0_blk0, y0_blk0, z), (time, x0_blk0, y, z), (time, x, y, z), (), (y0_blk0, x)]
`
to
`
{(time, t), (t, x, y, z), (time, x0_blk0, y, z), (t, x0_blk0, y0_blk0, z), (t, x0_blk0, y, z), (time, x0_blk0, y0_blk0, z), (x0_blk0, x), (y0_blk0, x), (y0_blk0, y), (time, x, y, z)}
`

Reply:
The less unnecessary code I see around, the happier I am